pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1491,14 +1491,15 @@ public CompletableFuture<Void> close(
         shadowReplicators.forEach((__, replicator) -> futures.add(replicator.disconnect()));
         if (disconnectClients) {
             futures.add(ExtensibleLoadManagerImpl.getAssignedBrokerLookupData(
-                    brokerService.getPulsar(), topic).thenAccept(lookupData ->
-                    producers.values().forEach(producer -> futures.add(producer.disconnect(lookupData)))
+                    brokerService.getPulsar(), topic).thenAccept(lookupData -> {
+                      producers.values().forEach(producer -> futures.add(producer.disconnect(lookupData)));
+                      subscriptions.forEach((s, sub) -> futures.add(sub.disconnect(lookupData)));
```

Comment:
We need to confirm the dispatching behavior when we close the topic with disconnectClients=false.

I think our expectation is to stop dispatching messages to consumers when the ledger is fenced. Could we confirm if this happens correctly? (also we probably want to add some metrics to show dispatcher pause time too. This can be a separate PR too.)